lib/Interpreter/DynamicLibraryManager.cpp

```diff
@@ -10,179 +10,55 @@
+        SplitPaths(Env, CurPaths, utils::kPruneNonExistant, platform::kEnvDelim,
+                   Opts.Verbose());
+        for (const auto& Path : CurPaths)
+          m_SystemSearchPaths.push_back(Path.str());
       }
-      initialized = true;
     }
 
-    for (std::vector<std::string>::const_iterator I = SysPaths.begin(),
-           E = SysPaths.end(); I != E; ++I)
-      Paths.push_back((*I).c_str());
-#endif
-  }
-#elif defined(LLVM_ON_WIN32)
-  static void GetSystemLibraryPaths(llvm::SmallVectorImpl<std::string>& Paths) {
-    char buff[MAX_PATH];
-    // Generic form of C:\Windows\System32
-    HRESULT res =  SHGetFolderPathA(NULL,
-                                    CSIDL_FLAG_CREATE | CSIDL_SYSTEM,
-                                    NULL,
-                                    SHGFP_TYPE_CURRENT,
-                                    buff);
-    if (res != S_OK) {
-      assert(0 && "Failed to get system directory");
-      return;
-    }
-    Paths.push_back(buff);
-
-    // Reset buff.
-    buff[0] = 0;
-    // Generic form of C:\Windows
-    res =  SHGetFolderPathA(NULL,
-                            CSIDL_FLAG_CREATE | CSIDL_WINDOWS,
-                            NULL,
-                            SHGFP_TYPE_CURRENT,
-                            buff);
-    if (res != S_OK) {
-      assert(0 && "Failed to get windows directory");
-      return;
-    }
-    Paths.push_back(buff);
-  }
+    platform::GetSystemLibraryPaths(m_SystemSearchPaths);
 
-  static bool is_dll(const char *filename) {
-    bool is_dll = false;
-    HANDLE hFile = CreateFile(filename, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0);
-    HANDLE hFileMapping = CreateFileMapping(hFile, NULL, PAGE_READONLY, 0, 0, NULL);
-    if (hFileMapping == INVALID_HANDLE_VALUE) {
-      CloseHandle(hFile);
-      return false;
-    }
-    LPVOID lpFileBase = MapViewOfFile(hFileMapping, FILE_MAP_READ, 0, 0, 0);
-    if (lpFileBase == NULL) {
-      CloseHandle(hFileMapping);
-      CloseHandle(hFile);
-      return false;
-    }
-    PIMAGE_DOS_HEADER pDOSHeader = static_cast<PIMAGE_DOS_HEADER>(lpFileBase);
-    if (pDOSHeader->e_magic == IMAGE_DOS_SIGNATURE) {
-      PIMAGE_NT_HEADERS pNTHeader = reinterpret_cast<PIMAGE_NT_HEADERS>((PBYTE)lpFileBase + pDOSHeader->e_lfanew);
-      if ((pNTHeader->Signature == IMAGE_NT_SIGNATURE) &&
-          ((pNTHeader->FileHeader.Characteristics & IMAGE_FILE_DLL)))
-        is_dll = true;
-    }
-    UnmapViewOfFile(lpFileBase);
-    CloseHandle(hFileMapping);
-    CloseHandle(hFile);
-    return is_dll;
-  }
-#else
-# error "Unsupported platform."
-#endif
-
-}
-
-namespace cling {
-  DynamicLibraryManager::DynamicLibraryManager(const InvocationOptions& Opts)
-    : m_Opts(Opts), m_Callbacks(0) {
-    GetSystemLibraryPaths(m_SystemSearchPaths);
+    // This will currently be the last path searched, should it be pushed to
+    // the front of the line, or even to the front of user paths?
```

Comment:
I agree - this should end up very early, before $LD_LIBRARY_PATH and system oaths. I don't care about the order wrt user provided ones.